extensions/amp-analytics/0.1/amp-analytics.js

```diff
@@ -358,7 +358,11 @@ export class AmpAnalytics extends AMP.BaseElement {
               trigger['selector'] = this.element.parentElement.tagName;
               trigger['selectionMethod'] = 'closest';
               this.addTrigger_(trigger);
-            } else if (trigger['selector']) {
+            } else if (
+              trigger['selector'] &&
+              startsWith(trigger['selector'], '${') &&
+              endsWith(trigger['selector'], '}')
+            ) {
```

Comment:
Just to be clear, this could break existing configs if they're using analytics-only variables ($IF, $MATCH $NOT, etc), or if the expansion is nested within text:
```
"selector": "my-class-$FRAGMENT_PARAM(classId,0)"
```

And we believe the trade off is worth it?

Reply:
flame chart:
I'm only testing with uncompiled code, so it's not very accurate. But you many notice that the last long task showed took less time to complete. 

> Just to be clear, this could break existing configs if they're using analytics-only variables ($IF, $MATCH $NOT, etc)
We introduced the feature before analytics only variables. And we haven't documented those variables yet. So I think that won't be an issue. 

>or if the expansion is nested within text:
Yes, it will lead to breakage in this case. But I checked all vendor usage. They're all using `${selector}`. So I think it's good indication that we can proceed. 

If you are still concerned, I can check for `$` and throw an error message.

Reply:
Awesome, thanks for the clarification!

> If you are still concerned, I can check for $ and throw an error message.

Your call, but I'm good without the message.

Reply:
Cool. I'll proceed without the message : ) I could add a note that `selector` expansion is only supported for a single variable in the format of `${}`. But I also really want to deprecate this feature and am hesitate to document the support. 😢